dds/DCPS/DataWriterImpl_T.h

```diff
@@ -327,6 +327,13 @@ DataWriterImpl_T
   {
     const DDS::DataRepresentationIdSeq repIds =
       get_effective_data_rep_qos(qos_.representation.value, false);
+    if (qos_.representation.value.length() > 0 &&
+       (qos_.representation.value[0] == DDS::XCDR_DATA_REPRESENTATION ||
+        qos_.representation.value[0] == DDS::XCDR2_DATA_REPRESENTATION))
```

Comment:
I know we don't support it, but `XML_DATA_REPRESENTATION` also uses encapsulation. That could also be added, but it would be simpler if we could assume that if `representation.value` isn't empty then we need encapsulation.